targets/TARGET_Cypress/TARGET_PSOC6_FUTURE/device/drivers/include/cy_ipc_drv.h

```diff
@@ -615,11 +615,13 @@ __STATIC_INLINE void  Cy_IPC_Drv_SetInterrupt(IPC_INTR_STRUCT_Type* base, uint32
 *******************************************************************************/
 __STATIC_INLINE void  Cy_IPC_Drv_ClearInterrupt(IPC_INTR_STRUCT_Type* base, uint32_t ipcReleaseMask, uint32_t ipcNotifyMask)
 {
+    uint32_t unused;
```

Comment:
i think you should add volatile so the compiler wont optimize that out

Reply:
I was thinking about it, but the ARM clang reference manual states (section 7.12 Qualifiers):
> Value computation of an lvalue expression with a volatile qualified type constitutes an access to the corresponding object, even when the value is discarded.

So I think it's ok, as it's hw register (data field), that is declared volatile. Variable `unused` could be skipped.
Same holds true for GCC.